testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java

```diff
@@ -616,7 +616,7 @@ private static void closeUnchecked(AutoCloseable closeable)
     {
         private Session defaultSession;
         private int nodeCount = 3;
-        private Map<String, String> extraProperties = new HashMap<>();
+        private Map<String, String> extraProperties = ImmutableMap.of();
```

Comment:
I do not get this change. What was wrong with previous appraoch?

Reply:
HashMap.put/putAll could potentially silently overwrite existing value. This is not a case with ImmutableMap

Reply:
Oh I see. Just do `putIfAbsent` and assert it returns `null`. Code will be much simpler.

Reply:
Despite 
>  Code will be much simpler.

This is a step backward from Immutable collections, which relaxes the code style - opens a possibility to override properties without intention. I'd prefer to left it as it is.

Reply:
I do not see it as such. It is not that ImmutableMap prevents you from overriding properties,  but the logic in `addProperties`/`addProperty`. 
Yet, whatever I will not fight over it.

Reply:
I agree with @losipiuk. Even if you have ImmutableMap you're still using a builder to construct it in those methods and anyone can override there if needed.

The safety comes from the fact that the only way to set these properties is via `addProperties/addProperty` and as long as those methods prevent overriding there's no possibility of overriding happening by mistake.

I also don't think it's worth arguing over even though the code looks much simpler and self-documenting with the change applied.